transposon/genotypeClones.py

```diff
@@ -523,17 +523,33 @@ def assignToNodes(G, key, annotationdict, default=None):
     print("[genotypeClones] added annotation to " + str(hadAnnotation) + " nodes; " + str(missingAnnotation) + " missing were assigned \"" + str(default) + "\".", file=sys.stderr)
 
 
-## Convert bipartite graph into a Jaccard index graph of the cells
+## Convert bipartite graph into a weighted Jaccard index graph of the cells
 def toJaccard(G):
     def jaccard(G, u, v):
-        unbrs = set(G[u])
-        vnbrs = set(G[v])
-        return float(len(unbrs & vnbrs)) / len(unbrs | vnbrs)
-    
+        u_bcs = set(G[u])
+        v_bcs = set(G[v])
+        common_bcs = u_bcs & v_bcs
+        common_u_weight = sum([G.edges[(u, bc)]["weight"] for bc in common_bcs])
+        common_v_weight = sum([G.edges[(v, bc)]["weight"] for bc in common_bcs])
+        return float((common_u_weight + common_v_weight) / (G.nodes[u]["weight"] + G.nodes[v]["weight"]))
     cells = [n for n in G.nodes if G.nodes[n]['type'] == 'cell']
     return nx.bipartite.generic_weighted_projected_graph(G, cells, weight_function=jaccard)
 
 
+## Unlink cells with low similarity (Jaccard-index)
+def pruneLowJaccard(G, min_jaccard):
+    J = toJaccard(G)
+    low_jaccard = [e for e in J.edges if J.edges[e]["weight"] < min_jaccard]
+    edges_to_remove = set()
+    for edge in low_jaccard:
+        u, v = edge
+        bcs = set(G[u]) & set(G[v])
+        edges_to_remove.update([(v, bc) if (G.edges[(u, bc)]["weight"] > G.edges[(v, bc)]["weight"]) else (u, bc) for bc in bcs])
```

Comment:
what's this doing?

Reply:
it is integrating over all pairs of cells (`u`, `v`) with low Jaccard index, identifying their shared bcs and selecting for each bc connection to which cell to remove (the one with lowest weight).

Reply:
add that as a comment

Reply:
done